File: spark-in-action/python/CsvToDataframeApp.py
import os
from pyspark.sql import SparkSession

curr_dir = os.path.dirname(__file__)
relative_path = os.path.join("..", "..", "..", "..", "data", "books.csv")
absolute_path = os.path.join(curr_dir, relative_path)

# create a session on a local master
session = SparkSession.builder
	.appName("CSV to Dataset")
	.master("local[*]")
	.getOrCreate()

# reads a CSV file with header and stores it in a dataframe
df = session.read.csv(header=True, inferSchema=True, path=absolute_path)

# show at most 5 rows from the dataframe
df.show(5)

# stop SparkSession at the end of the application
session.stop()

File: spark-in-action/scala/src/main/scala/CsvToDataframeApp.scala
import org.apache.spark.sql.SparkSession

object CsvToDataframeApp {
  def main(args: Array[String]): Unit = {

    // Creates a session on a local master
    val spark = SparkSession.builder
      .appName("CSV to Dataset")
      .master("local[*]")
      .getOrCreate

    // Reads a CSV file with header, called books.csv, stores it in a dataframe
    val df = spark.read.format("csv")
      .option("header", "true")
      .load("data/books.csv")

    // Shows at most 5 rows from the dataframe
    df.show(5)

    // Good to stop SparkSession at the end of the application
    spark.stop
  }

}